c10/core/Device.cpp

```diff
@@ -9,6 +9,21 @@
 #include <string>
 #include <tuple>
 #include <vector>
+#include <regex>
+
+// Test below is adapted from https://stackoverflow.com/a/41186162
```

Comment:
nit: might be nice to prefix this with "Check if compiler has working std::regex implementation", but reading the conditional works too.

Reply:
done!